In [19]:
import pandas as pd
from ingredients_lists import *

In [2]:
data = pd.read_csv('ingredientes_estandarizados.csv')

In [3]:
df = data.copy()

In [20]:
df

receta  \
0                  Iogurt grec amb fraules   
1                  Iogurt grec amb fraules   
2                  Iogurt grec amb fraules   
3                  Iogurt grec amb fraules   
4                  Iogurt grec amb fraules   
...                                    ...   
1058  Pollastre al forn a les fines herbes   
1059  Pollastre al forn a les fines herbes   
1060  Pollastre al forn a les fines herbes   
1061  Pollastre al forn a les fines herbes   
1062          Broqueta de gambes amb xoriç   

                    ingrediente_original  cantidad unidad  \
0                       Iogurt grec 200g       200      g   
1                       Fraules 100-150g       150      g   
2                    Nous i ametlles 30g        30      g   
3                             Xía 10-20g        20      g   
4                       Canyella al gust         0    NaN   
...                                  ...       ...    ...   
1058    5g d'orenga fresc (0.5g en pols)         0    NaN   
1059               15g de mostassa Dijon         0    NaN   
1060  15ml d'oli d'oliva (una cullerada)         0    NaN   
1061                 Sal i pebre al gust         0    NaN   
1062                          llagostins       250      g   

     ingrediente_estandar       tipo_ingrediente  
0             iogurt grec      Làctics i greixos  
1                 fraules                Fruites  
2         nous i ametlles  Fruites secs i altres  
3                     xía  Fruites secs i altres  
4                canyella             Condiments  
...                   ...                    ...  
1058         orenga fresc             Condiments  
1059       mostassa dijon             Condiments  
1060          oli d'oliva             Condiments  
1061          sal i pebre             Condiments  
1062           llagostins    Carns, peixos i ous  

[1063 rows x 6 columns]

In [5]:
df.columns

Index(['receta', 'ingrediente_original', 'cantidad', 'unidad',
       'ingrediente_estandar', 'tipo_ingrediente'],
      dtype='object')

In [6]:
df.shape

(1063, 6)

In [7]:
df['ingrediente_estandar'].nunique()

231

### Líneas para manipular!

In [21]:
for index, row in df.iterrows():
    if row['ingrediente_estandar'] in Carns_peixos_i_ous:
        df.at[index, 'tipo_ingrediente'] = 'Carns, peixos i ous'
    else:
        df.at[index, 'tipo_ingrediente'] = row['tipo_ingrediente']

In [22]:
df['tipo_ingrediente'].value_counts()

tipo_ingrediente
Condiments               319
Verdures                 274
Fruites secs i altres    148
Carns, peixos i ous      140
Làctics i greixos        119
Fruites                   55
Name: count, dtype: int64

In [10]:
df.sort_values(by=['ingrediente_estandar', 'cantidad'], ascending=[True, True])

receta  \
14   Solomillo de vedella Strogonoff i arròs basmati   
27                                Cuinat de llenties   
74                                         Vinagreta   
272  Ous bullits amb salmó fumat, tàperes i maionesa   
608                                        Pa de lli   
..                                               ...   
603                                        Pa de lli   
672                                      Cetogranola   
3                            Iogurt grec amb fraules   
733                Iogurt grec amb ametlles i nabius   
898    Pudding de xía amb fruits vermells i ametlles   

            ingrediente_original  cantidad unidad ingrediente_estandar  \
14                      Agua 10g         0    NaN                aigua   
27                 750ml d'aigua         0    NaN                aigua   
74                 10 ml d'aigua         0    NaN                aigua   
272                        Aigua         0    NaN                aigua   
608        120 g d’aigua calenta         0    NaN                aigua   
..                           ...       ...    ...                  ...   
603        xia (una culleradeta)        10      g                  xía   
672     5-10g de chía (opcional)        10      g                  xía   
3                     Xía 10-20g        20      g                  xía   
733               10-20g de chía        20      g                  xía   
898  3 cullerades soperes de xía        30      g                  xía   

          tipo_ingrediente  
14   Fruites secs i altres  
27   Fruites secs i altres  
74   Fruites secs i altres  
272  Fruites secs i altres  
608  Fruites secs i altres  
..                     ...  
603  Fruites secs i altres  
672  Fruites secs i altres  
3    Fruites secs i altres  
733  Fruites secs i altres  
898  Fruites secs i altres  

[1063 rows x 6 columns]

In [23]:
df[df['ingrediente_estandar'].str.contains("ous", na=False)]

receta  \
2                               Iogurt grec amb fraules   
43               Ous frits amb pernil salat i ½ alvocat   
88    Hamburguesa de salmó amb pernil salat i patata...   
134                                 Amanida de sardines   
136             Frittata d’espinacs i formatge de cabra   
167                                       Amanida grega   
168                 Ous remenats amb formatge i cibulet   
182                 Pa de plàtan amb xispes de xocolata   
193                 Pa de plàtan amb xispes de xocolata   
220                        Fritatta de xerris amb xoriç   
255                                             Granola   
262                                             Granola   
271     Ous bullits amb salmó fumat, tàperes i maionesa   
302              Tortilla de coliflor, formatge i salmó   
339             Ous remenats o fregits amb cuixot salat   
360                Frittata de bròquil, cheddar i xoriç   
377               Ous remenats amb formatge i espàrrecs   
395                                          Ceto-creps   
427                    Ous a la planxa amb pernil salat   
440                             Amanida de tonyina i ou   
441              Ous remenats amb formatge i xampinyons   
456                     Tortilla de mozzarella i xerris   
469                     Muffins de nabius (sense sucre)   
475                                          Ceto-Pizza   
486                                Coquitos de xocolata   
498                        Frittata de xerris amb xoriç   
542              Amanida d’espinacs ruca, nous i nabius   
562     Tortitas de coliflor amb formatge i salmó fumat   
572                                  Brioix de xocolata   
605                                           Pa de lli   
606                                           Pa de lli   
623                                    Truita de gambes   
656                   Verdures bullides amb ous bullits   
657   Ous bullits ½ alvocat una mica de salmó fumat ...   
670                                         Cetogranola   
677                                         Cetogranola   
693                                        Truita-pizza   
702                         Crep amb xocolata i fraules   
732                   Iogurt grec amb ametlles i nabius   
755                                           Ous poché   
758            Frittata de salmó, espinacs i xampinyons   
775                                            Pad-thai   
789                                Fajitas de pollastre   
793                                             Lasanya   
846                   Truita d’espàrrecs, all i cúrcuma   
878                                  Truita de carxofes   
881                       Amanida d’escarola amb bolets   
895                                           Sopa miso   
926                       Pa (lli) amb pernil i alvocat   
934            Roses de pollastre amb pernil i formatge   
947                                      Pastís de carn   
952       Truita d'espinacs amb formatge blau i cúrcuma   
971                        Dorada a la sal amb maionesa   
978          Truita de Xampinyons amb Formatge de Cabra   
999                 Ous Remenats amb Espinacs i Alvocat   
1018                Truita amb formatge blau i espinacs   
1039                Pebrots farcits de peix i formatge.   
1042              Ou bullit amb salmó fumat i ½ alvocat   
1047            Ous remenats o fregits amb cuixot salat   

                                   ingrediente_original  cantidad unidad  \
2                                   Nous i ametlles 30g        30      g   
43                                                3 ous         3    NaN   
88                                                 1 ou         1    NaN   
134                             10g de nous (o pinyons)        10      g   
136                                               6 ous         6    NaN   
167                             10g de nous (o pinyo

In [12]:
mask = df[(df['ingrediente_estandar'].str.contains("dents", na=False)) & (df['cantidad'] == 3)]

In [13]:
mask = df[df['ingrediente_estandar'].str.contains("bac", na=False)]

In [14]:
#df.loc[mask.index, 'ingrediente_estandar'] = 'bacallà'
#df.loc[mask.index, 'cantidad'] = None
#df.loc[mask.index, 'unidad'] = None

In [15]:
#df.loc[[186],'tipo_ingrediente'] = 'Altres'
#df.loc[[645],'cantidad'] = 1
#df.loc[[781],'unidad'] = None

n_fila = [{
    'receta': 'Broqueta de gambes amb xoriç',
    'ingrediente_original': 'llagostins',
    'cantidad': 250,
    'unidad': 'g',
    'ingrediente_estandar': 'llagostins'
}]

df = pd.concat([df, pd.DataFrame(n_fila)], ignore_index=True)

In [16]:
#df['ingrediente_estandar'] = df['ingrediente_estandar'].replace('salmo fresc', 'salmó fresc')
#df['ingrediente_estandar'] = df['ingrediente_estandar'].replace('romani fresc', 'romaní fresc')


In [24]:
df.to_csv('ingredientes_estandarizados.csv', index=False)